### PR TITLE
URO-82 URO-143 Connect search

### PR DESCRIPTION
**Pull request by dakotablair**
- [ ] [URO-82](https://kbase-jira.atlassian.net/browse/URO-82) Create Navigator Component
  - [x] Align (roughly) to mockup [URO-78: Create Mockup for Narrative Navigator](https://kbase-jira.atlassian.net/browse/URO-78)
  - [ ] Implement user behaviors existing in previous Navigator
    - [x] Manages state for searches and filters
    - [x] Manages quantity of search results displayed
    - [ ] Manages state for the control menu including
      - [ ] delete
      - [ ] copy
      - [ ] link (to an organization)
      - [ ] rename
      - [ ] share
    - [ ] Handles narrative metadata changes at least to the standard of the previous Navigator
  - [ ] Write tests with 100% coverage
- [x] [URO-143](https://kbase-jira.atlassian.net/browse/URO-143) Connect Navigator to search api
  - [x] Navigator displays narratives matching chosen filters loaded from [searchapi2](https://github.com/kbase/navigator/blob/main/src/client/utils/searchNarratives.ts) using RTK Query.
  - [x] Write tests with 100% coverage on new files
    - [x] `src/common/api/searchApi.ts`
    - [x] `src/features/navigator/hooks.ts`

[URO-82]: https://kbase-jira.atlassian.net/browse/URO-82
[URO-143]: https://kbase-jira.atlassian.net/browse/URO-143

### CODE REVIEW

File: src/features/navigator/RefreshButton.tsx [hunk 1]
```diff
@@ -1,20 +1,30 @@
...
 import { FontAwesomeIcon as FAIcon } from '@fortawesome/react-fontawesome';
 import { Button } from '../../common/components';
+import type { AppDispatch } from '../../app/store';
 import { useAppDispatch } from '../../common/hooks';
-import { setFresh } from './navigatorSlice';
+import { clearCacheAction } from '../../common/api/searchApi';
 import classes from './Navigator.module.scss';
 
-const RefreshButton: FC = () => {
+// This handler factory pattern is primarily to aid unit testing.
```
Comment (dakotablair) [inline src/features/navigator/RefreshButton.tsx:10]: [Use](https://stackoverflow.com/questions/39755439/how-to-mock-imported-named-function-in-jest-when-module-is-unmocked/53402206#53402206) `jest.requireActual(moduleName)` instead to make this code simpler.

File: src/features/navigator/hooks.ts [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+    },
+    sorts: [SearchAPISort[sort], ['_score', 'desc']],
+    types: ['KBaseNarrative.Narrative'],
+  };
+};
+
+export const useNarratives = (params: getNarrativesParams) => {
+  const dispatch = useAppDispatch();
+  const narrativesPrevious = useAppSelector(narratives);
+  const searchQuery = makeGetNarrativesParams(params);
```
Comment (dakotablair) [inline src/features/navigator/hooks.ts:85]: Memoize these parameters using `useMemo`, see [this example](https://github.com/kbase/ui-refresh-test/blob/225a388eb6e5f4d2dc7aaae2e1224ec8c313ed3c/src/features/collections/data_products/TaxaCount.tsx#L14-L15).

File: src/features/navigator/RefreshButton.tsx (hunk 1 above)
Comment (dakotablair) [reply src/features/navigator/RefreshButton.tsx]: Actually, `<RefreshButton />` does not need to accept a separate handler at this time in any case.

**Review by dauglyon — APPROVED** [review]
This all lgtm!

**Pull request merged**